### ISSUE TITLE
Native Image tests failing with NoClassDefFoundError

### ISSUE THREAD

**Issue by mpeddada1** [googleapis/java-spanner-jdbc#910]
This was observed in https://github.com/googleapis/java-spanner-jdbc/pull/898

```
Caused by: java.lang.NoClassDefFoundError: com/google/cloud/spanner/IntegrationTestEnv
	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/java.net.URLClassLoader.defineClass(URLClassLoader.java:555)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:458)
	at java.base/java.net.URLClassLoader$1.run(URLClassLoader.java:452)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:451)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.getDeclaredMethods0(Native Method)
	at java.base/java.lang.Class.privateGetDeclaredMethods(Class.java:3166)
	at java.base/java.lang.Class.privateGetPublicMethods(Class.java:3191)
	at java.base/java.lang.Class.privateGetPublicMethods(Class.java:3197)
	at java.base/java.lang.Class.getMethods(Class.java:1904)
	at org.junit.platform.commons.util.ReflectionUtils.getDefaultMethods(ReflectionUtils.java:1518)
	at org.junit.platform.commons.util.ReflectionUtils.getDeclaredMethods(ReflectionUtils.java:1491)
	at org.junit.platform.commons.util.ReflectionUtils.findAllMethodsInHierarchy(ReflectionUtils.java:1433)
	at org.junit.platform.commons.util.ReflectionUtils.findMethods(ReflectionUtils.java:1417)
	at org.junit.platform.commons.util.ReflectionUtils.findMethods(ReflectionUtils.java:1403)
	at org.junit.vintage.engine.descriptor.TestSourceProvider.lambda$findMethod$1(TestSourceProvider.java:75)
	at java.base/java.util.HashMap.computeIfAbsent(HashMap.java:1134)
	at java.base/java.util.Collections$SynchronizedMap.computeIfAbsent(Collections.java:2682)
	at org.junit.vintage.engine.descriptor.TestSourceProvider.findMethod(TestSourceProvider.java:75)
	at org.junit.vintage.engine.descriptor.TestSourceProvider.computeTestSource(TestSourceProvider.java:56)
	at java.base/java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1705)
	at org.junit.vintage.engine.descriptor.TestSourceProvider.findTestSource(TestSourceProvider.java:47)
	at org.junit.vintage.engine.discovery.RunnerTestDescriptorPostProcessor.addChildrenRecursively(RunnerTestDescriptorPostProcessor.java:62)
	at org.junit.vintage.engine.discovery.RunnerTestDescriptorPostProcessor.applyFiltersAndCreateDescendants(RunnerTestDescriptorPostProcessor.java:41)
	at org.junit.vintage.engine.discovery.VintageDiscoverer.discover(VintageDiscoverer.java:46)
	at org.junit.vintage.engine.VintageTestEngine.discover(VintageTestEngine.java:64)
	at org.junit.platform.launcher.core.EngineDiscoveryOrchestrator.discoverEngineRoot(EngineDiscoveryOrchestrator.java:152)
	... 18 more
Caused by: java.lang.ClassNotFoundException: com.google.cloud.spanner.IntegrationTestEnv
	at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:476)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:589)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 50 more
```

**Comment by mpeddada1** [comment]
As discovered in #912, the issue was first introduced with shared-config 1.5.1 where the native-maven-plugin was upgraded to 0.9.12. Downgrading to 1.5.0 helped address the failure but still need troubleshoot further to see what changed in native-maven-plugin. Still need to keep this issue open.

**Comment by mpeddada1** [comment]
This is an issue that was recently introduced with native-maven-plugin 0.9.12 where a dependency of type `test-jar` isn't being included in the classpath that is used with the `native-image` command: https://github.com/mpeddada1/0.9.12-test-jar

**Comment by mpeddada1** [comment]
This issue was resolved starting with [native-maven-plugin 0.9.14](https://github.com/graalvm/native-build-tools/releases/tag/0.9.14). It was brought in by [shared-config 1.5.4](https://github.com/googleapis/java-shared-config/pull/511). Created follow-up to remove workaround: https://github.com/googleapis/java-spanner-jdbc/pull/1025
